### ISSUE TITLE
[java] UnnecessaryImport false-positive on on-demand import whose package supplies a referenced type

### ISSUE THREAD

**Issue by Emilyaxe** [pmd/pmd#6626]
**Affects PMD Version:**  7.23.0

**Rule:** 

Please provide the rule name and a link to the rule documentation:
UnnecessaryImport — https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport

**Description:**
PMD incorrectly reports a wildcard import (`import p.*;`) as unused, even though a type from that package is actually referenced. In the example, `C` is used as `C.V` and is resolved via `import p.*;`. Removing this import would cause a compilation error, so it is clearly required.

In contrast, PMD correctly handles the equivalent single-type import (`import p.C;`). This shows that the rule treats semantically equivalent imports inconsistently.

**Code Sample demonstrating the issue:**

```java
// package p (provides type C with constant V)
package p;
public class C {
    public static final String V = "v";
}
```
```java
// not flagged (correct)
package demo.before;

import p.C;

public class U {
    private void g() {
        String k = C.V;
    }
}
```
```java
// UnnecessaryImport reported on line 3 (false-positive)
package demo.after;

import p.*;          // <-- PMD: "Unused import 'p.*'"

public class U {
    private void g() {
        String k = C.V;   // C is resolved through p.*
    }
}
```

**Expected outcome:** 
No `UnnecessaryImport` violation, because removing `import p.*;` would make `C` unresolvable.

**Actual outcome:** PMD reports `Unused import 'p.*'` on line 3, despite `C` from package `p` being referenced in the class body.

**Running PMD through:** *[CLI ]*

**Comment by UncleOwen** [comment]
Please make sure to provide the appropriate `--aux-classpath`:
```
$ pmd-7.23.0 check -R category/java/codestyle.xml/UnnecessaryImport -d src/
[WARN] Progressbar rendering conflicts with reporting to STDOUT. No progressbar will be shown. Try running with argument -r <file> to output the report to a file instead.
[WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.23.0/pmd_userdocs_incremental_analysis.html
src/main/java/demo/after/U.java:4:      UnnecessaryImport:      Unused import 'p.*'
$ pmd-7.23.0 check -R category/java/codestyle.xml/UnnecessaryImport -d src/ --aux-classpath=target/classes/
[WARN] Progressbar rendering conflicts with reporting to STDOUT. No progressbar will be shown. Try running with argument -r <file> to output the report to a file instead.
[WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.23.0/pmd_userdocs_incremental_analysis.html
```

**Comment by Emilyaxe** [comment]
> Please make sure to provide the appropriate `--aux-classpath`:
> 
> ```
> $ pmd-7.23.0 check -R category/java/codestyle.xml/UnnecessaryImport -d src/
> [WARN] Progressbar rendering conflicts with reporting to STDOUT. No progressbar will be shown. Try running with argument -r <file> to output the report to a file instead.
> [WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.23.0/pmd_userdocs_incremental_analysis.html
> src/main/java/demo/after/U.java:4:      UnnecessaryImport:      Unused import 'p.*'
> $ pmd-7.23.0 check -R category/java/codestyle.xml/UnnecessaryImport -d src/ --aux-classpath=target/classes/
> [WARN] Progressbar rendering conflicts with reporting to STDOUT. No progressbar will be shown. Try running with argument -r <file> to output the report to a file instead.
> [WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.23.0/pmd_userdocs_incremental_analysis.html
> ```

Thank you, I will carefully study the official document.

**Comment by UncleOwen** [comment]
Closing as not reproduceable
